src/main/java/com/faforever/client/teammatchmaking/TeamMatchmakingMapListController.java

```diff
@@ -0,0 +1,198 @@
+import javafx.scene.layout.FlowPane;
+import javafx.scene.layout.Pane;
+
+import com.faforever.client.domain.server.MatchmakerQueueInfo;
+import com.faforever.client.fx.NodeController;
+import com.faforever.client.map.MapService;
+import javafx.scene.layout.VBox;
+import javafx.stage.Screen;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+import reactor.core.publisher.Flux;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.SortedMap;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+
+@Slf4j
+@Component
+@Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+@RequiredArgsConstructor
+
+public class TeamMatchmakingMapListController extends NodeController<Pane> {
+
+  public Pane root;
+  public FlowPane tilesContainer;
+  private final MapService mapService;
+  private final UiService uiService;
+  private final I18n i18n;
+  private final PlayerService playerService;
+  private final FxApplicationThreadExecutor fxApplicationThreadExecutor;
+  private final int TILE_SIZE = 125;
+  public ScrollPane scrollContainer;
+  public VBox loadingPane;
+  private Map<MatchmakerQueueMapPool, List<MapVersion>> rawBrackets;
+  private SortedMap<MatchmakerQueueMapPool, List<MapVersion>> sortedBrackets;
+  private SortedMap<MatchmakerQueueMapPool, List<MapVersion>> sortedBracketsWithDuplicates;
+  private Integer playerBracketIndex = null;
+  private Pane dialogRoot;
+
+
+  @Override
+  protected void onInitialize() {
+    JavaFxUtil.bindManagedToVisible(loadingPane);
+  }
+
+  @Override
+  public Pane getRoot() {
+    return root;
+  }
+
+
+  public void init(MatchmakerQueueInfo queue, Pane dialogRoot) {
+    this.dialogRoot = dialogRoot;
+    mapService.getMatchmakerBrackets(queue).subscribe(rawBrackets -> {
+      loadingPane.setVisible(false);
+      this.rawBrackets = rawBrackets;
+      this.sortedBracketsWithDuplicates = this.getSortedBrackets(rawBrackets);
+      this.sortedBrackets = this.removeDuplicates(this.getSortedBrackets(rawBrackets));
+
+      PlayerInfo player = playerService.getCurrentPlayer();
+      LeaderboardRating ratingBean = player.getLeaderboardRatings().get(queue.getLeaderboard().technicalName());
+      double rating = ratingBean.mean() - 3 * ratingBean.deviation();
+      this.playerBracketIndex = this.getPlayerBracketIndex(this.sortedBrackets, rating);
+
+      List<MapVersion> values = this.sortedBrackets.values().stream().flatMap(List::stream).toList();
+      this.resizeToContent(values.size(), this.TILE_SIZE);
+
+      Flux.fromIterable(values)
+          .publishOn(fxApplicationThreadExecutor.asScheduler())
+          .doOnNext(this::addMapTile)
+          .subscribe();
```

Comment:
This doesn't need to be a flux you can just run this in the fx application thread executor.

And really this parent flux should be published on the fx application executor so this is unnecessary.